tst/ProducerCallbackProviderApiTest.cpp

```diff
@@ -42,53 +43,93 @@ TEST_F(ProducerCallbackProviderApiTest, AddProducerCallbackProvider_Returns_Vali
-        {"ap-east-1", "https://kinesisvideo.ap-east-1.amazonaws.com"},
-        {"af-south-1", "https://kinesisvideo.af-south-1.amazonaws.com"},
-        {"us-iso-east-1", "https://kinesisvideo-fips.us-iso-east-1.c2s.ic.gov"},
-        {"us-iso-west-1", "https://kinesisvideo-fips.us-iso-west-1.c2s.ic.gov"},
-        {"us-isob-east-1", "https://kinesisvideo-fips.us-isob-east-1.sc2s.sgov.gov"},
-        {"us-gov-west-1", "https://kinesisvideo-fips.us-gov-west-1.amazonaws.com"},
-        {"us-gov-east-1", "https://kinesisvideo-fips.us-gov-east-1.amazonaws.com"},
-        {"cn-north-1", "https://kinesisvideo.cn-north-1.amazonaws.com.cn"},
-        {"cn-northwest-1", "https://kinesisvideo.cn-northwest-1.amazonaws.com.cn"},
-    };
-
-    for (auto it : regionToControlPlaneUrlMap) {
-        EXPECT_EQ(STATUS_SUCCESS,
-                  createDefaultCallbacksProvider(TEST_DEFAULT_CHAIN_COUNT, TEST_ACCESS_KEY, TEST_SECRET_KEY, TEST_SESSION_TOKEN,
-                                                 TEST_STREAMING_TOKEN_DURATION, (PCHAR) it.first.c_str(), TEST_CONTROL_PLANE_URI, mCaCertPath, NULL,
-                                                 TEST_USER_AGENT, API_CALL_CACHE_TYPE_NONE, TEST_CACHING_ENDPOINT_PERIOD, TRUE, &pClientCallbacks));
-
-        PCallbacksProvider pCallbacksProvider = (PCallbacksProvider) pClientCallbacks;
-        PCurlApiCallbacks pCurlApiCallbacks = (PCurlApiCallbacks) pCallbacksProvider->pApiCallbacks[0].customData;
-
-        EXPECT_EQ(0, STRNCMP(pCurlApiCallbacks->controlPlaneUrl, (PCHAR) it.second.c_str(), MAX_URI_CHAR_LEN));
-
-        EXPECT_EQ(STATUS_SUCCESS, freeCallbacksProvider(&pClientCallbacks));
-
-        EXPECT_EQ(STATUS_NULL_ARG, freeCallbacksProvider(NULL));
-    }
+    char buffer[MAX_URI_CHAR_LEN] = {0};
+    const ControlPlaneUrlTestParam& param = GetParam();
+
+    constructControlPlaneUrl(buffer, MAX_URI_CHAR_LEN, param.region.c_str(), param.isLegacyEndpoint);
+    EXPECT_STREQ(buffer, param.expectedUrl.c_str());
+}
+
+// Custom test name generator based on the region and legacy endpoint flag
+// Otherwise you'll see GeneratesCorrectUrls/0, GeneratesCorrectUrls/1, etc
+std::string ControlPlaneUrlTestName(const testing::TestParamInfo<ControlPlaneUrlTestParam>& info)
+{
+    const ControlPlaneUrlTestParam& param = info.param;
+
+    // Note: Gtest doesn't allow hyphens in the test names
+    // Replace hyphens with underscores to make the test name valid
+    std::string region = param.region;
+    std::replace(region.begin(), region.end(), '-', '_');
+
+    return region + "_" + (param.isLegacyEndpoint ? "Legacy" : "New");
 }
 
+// Instantiate the test suite with multiple test cases
+INSTANTIATE_TEST_SUITE_P(ControlPlaneUrlTests, ControlPlaneUrlTest,
+                         ::testing::Values(
+                             // Legacy endpoints
+                             ControlPlaneUrlTestParam{"us-east-1", "https://kinesisvideo.us-east-1.amazonaws.com", TRUE},
```

Comment:
Would it be better to group them by region or category, e.g., isolated regions, CN (China), etc.?

Reply:
I left the original unit test order:
1. Commercial regions
2. Govcloud/fips regions
3. China regions